op/model.py

```diff
@@ -72,6 +71,36 @@ def __init__(self, name, backend, cache):
 
         self._is_our_app = self.name == self._backend.app_name
 
+        self._status = None
+
+    @property
+    def status(self):
+        if not self._is_our_app:
+            return Unknown()
+
+        # TODO: do a leadership check (app status is not available to units for reading/writing).
+
+        if self._status:
+            return self._status
+
+        s = self._backend.status_get(True)
+        self._status = Status.from_string(s['status'], s['message'])
+
+        return self._status
+
+    @status.setter
+    def status(self, value):
+        if not isinstance(value, Status):
+            raise InvalidStatusError(f'Invalid value provided for application {self} status: {value}')
+
+        if not self._is_our_app:
+            raise RuntimeError(f'Unable to set status for a remote application {self}')
+
+        # TODO: do a leadership check (app status is not available to units for reading/writing).
```

Comment:
As above, this needs to be addressed now.

Reply:
Done.